designer/server/src/main/scala/pl/touk/nussknacker/ui/factory/DomainServices.scala

```diff
@@ -90,66 +96,63 @@ object DomainServices {
         deploymentManagersClassLoader
       )
 
-      actionServiceSupplier = new DelayedInitActionServiceSupplier
       additionalUIConfigProvider = AdditionalUIConfigProviderLoader.loadAdditionalUIConfigProvider(
         alreadyLoadedConfig,
         futureSttpBackend
       )
       // 1 hour is the delay to propagate all global notifications for all users
       globalNotificationRepository = InMemoryTimeseriesRepository[Notification](Duration.ofHours(1), Clock.systemUTC())
-      processingTypeDataProvider <- prepareProcessingTypeDataReload(
+      modelDataProvider <- prepareModelDataReload(
         designerConfigLoader,
         alreadyLoadedConfig,
-        infrastructureServices,
-        deploymentManagersClassLoader,
-        modelClassLoaderProvider,
         additionalUIConfigProvider,
-        actionServiceSupplier,
         globalNotificationRepository,
+        modelClassLoaderProvider
       )
-
-      feStatisticsRepository <- QuestDbFEStatisticsRepository.create(
-        actorSystem,
-        clock,
-        alreadyLoadedConfig.questDbSettings
-      )
-      countsReporter <- CountsReporterFactory.createCountsReporter(
-        alreadyLoadedConfig,
-        futureSttpBackend
+      // deployment data deps
+      actionServiceSupplier    = new DelayedInitActionServiceSupplier
+      actionRepository         = DbScenarioActionRepository.create(dbRef)
+      scenarioLabelsRepository = new ScenarioLabelsRepository(dbRef)
+      // TODO: get rid of Future based repositories - it is easier to use everywhere one implementation - DBIOAction based which allows transactions handling
+      futureProcessRepository = DBFetchingProcessRepository.createFutureRepository(
+        dbRef,
+        actionRepository,
+        scenarioLabelsRepository
       )
-      deploymentRepository = new DeploymentRepository(dbRef, clock)
-      deploymentsStatusesSynchronizer = new DeploymentsStatusesSynchronizer(
-        deploymentRepository,
-        processingTypeDataProvider.mapValues(
-          _.deploymentData.validDeploymentManagerOrStub.deploymentSynchronisationSupport
+      deploymentData <- DeploymentManagersLoader.load(
+        alreadyLoadedConfig.processingTypeConfigs(),
+        deploymentManagersClassLoader,
+        modelClassLoaderProvider,
+        modelDataProvider.mapValues(_.modelData),
+        getDeploymentManagerDependencies(
+          infrastructureServices,
+          _
         ),
-        dbioRunner
+        Some(
+          getSchedulingDependencies(
+            infrastructureServices,
+            actionServiceSupplier,
+            futureProcessRepository,
+            additionalUIConfigProvider,
+            _
+          )
+        )
       )
-      _ <- DeploymentsStatusesSynchronizationScheduler.resource(
-        actorSystem,
-        deploymentsStatusesSynchronizer,
-        alreadyLoadedConfig.deploymentsStatusesSynchronizationConfig
+      // ActionService initialization
```

Comment:
I changed the order a bit to make more clear:
- when ActionService is ready to use
- what services are based on `ProcessingTypeData`